### PR TITLE
feat: auto-identify users from OAuth proxy X-Forwarded-User header (#173)

### LINKED ISSUE #173: feat: Auto-identify users from OAuth proxy X-Forwarded-User header
## Problem

When the OAuth proxy sidecar is deployed (OpenShift `oauth-proxy`), users are already authenticated via Red Hat SSO before reaching JJI. However, JJI currently requires users to manually register by setting a `jji_username` cookie through the registration page. This creates a redundant step — users have already proven their identity through SSO.

## Proposed Solution

When JJI detects an `X-Forwarded-User` header (set by the OAuth proxy), it should:

1. **Skip manual registration** — automatically use the SSO identity as the JJI username
2. **Set the `jji_username` cookie** from `X-Forwarded-User` so downstream code works unchanged
3. **Bypass the registration page** — redirect `/register` to `/` when SSO identity is present

### Implementation Details

- Add a new middleware or extend `AuthMiddleware` to check for `X-Forwarded-User` header
- When present: set `jji_username` cookie to the header value (if not already set or if different)
- When absent: fall back to current cookie-based registration flow (backward compatible)
- Add a config option (e.g., `TRUST_PROXY_HEADERS=true`) to explicitly opt in — never trust forwarded headers without explicit configuration (security)
- The registration page should detect SSO users and either auto-redirect or show a "Logged in as X" message

### Security Considerations

- `X-Forwarded-User` must ONLY be trusted when `TRUST_PROXY_HEADERS` is enabled
- When enabled, the header should be stripped from external requests if not coming through the proxy (defense in depth)
- Document that this setting assumes a trusted reverse proxy (OAuth proxy sidecar)

## Context

The `jji-dev` deployment on CNV2 OpenShift now has an OAuth proxy sidecar configured:
- ServiceAccount: `jji-dev` with OAuth redirect annotation
- Route: `reencrypt` through proxy port 4180
- Skip-auth paths: `/health`, `/api/health`, `/metrics`
- Cookie: `_jji_oauth_proxy`

## Done

- [ ] Add `TRUST_PROXY_HEADERS` config option (default: `false`)
- [ ] Read `X-Forwarded-User` header in middleware when enabled
- [ ] Auto-set `jji_username` cookie from SSO identity
- [ ] Skip/redirect registration page for SSO-identified users
- [ ] Backward compatible — no behavior change when `TRUST_PROXY_HEADERS` is `false`
- [ ] Tests for both SSO and non-SSO flows
- [ ] Document the configuration in README

### PR DESCRIPTION
**Pull request by myakove**
Closes #173

## Features

- `TRUST_PROXY_HEADERS` config option (default: `false`)
- When enabled, reads `X-Forwarded-User` header and auto-sets `jji_username` cookie
- Skips/redirects registration page for SSO-identified users
- Backward compatible — no behavior change when disabled
- Tests for both SSO and non-SSO flows
- README documentation

## Security

- Header only trusted when `TRUST_PROXY_HEADERS=true` is explicitly set
- Assumes trusted reverse proxy (OAuth proxy sidecar)

## Testing

All tests pass.

## Summary by CodeRabbit

* **New Features**
  * Optional OAuth proxy/SSO support: auto-identify users from reverse-proxy identity headers, set a compatibility username cookie, and redirect SSO-identified users away from registration to the dashboard.

* **Configuration**
  * New toggle to enable trusting proxy-provided authentication headers; admin sessions and bearer tokens continue to take precedence.

* **Documentation**
  * README updated with setup steps, security guidance, and a sample proxy configuration.

* **Tests**
  * New tests validating proxy-header SSO behavior, redirects, precedence, and edge cases.

### CODE REVIEW

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

File: README.md [hunk 1]
```diff
@@ -69,6 +69,40 @@ Subscribe/unsubscribe is browser-only (managed via the web UI). To list users av
...
+When the header is absent, the standard cookie-based registration flow is used (backward compatible).
+
+### Example: OpenShift OAuth Proxy
+
+```yaml
+# In your Deployment, add the oauth-proxy sidecar:
+env:
+  - name: TRUST_PROXY_HEADERS
+    value: "true"
+```
```
[the remark this answers is not in the document]
Comment (myakove) [reply README.md:104]: Fixed as suggested.

File: tests/test_auth.py [hunk 2]
```diff
@@ -878,3 +878,128 @@ async def check_still_expired():
...
+        resp = proxy_client.get(
+            "/api/auth/me",
+            headers={"X-Forwarded-User": "sso-user"},
+            cookies=cookies,
+        )
+        assert resp.status_code == 200
+        data = resp.json()
+        assert data["username"] == "admin"
+        assert data["is_admin"] is True
+
```
[the remark this answers is not in the document]
Comment (myakove) [reply tests/test_auth.py:1006]: Fixed as suggested.

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

File: src/jenkins_job_insight/main.py [hunk 3]
```diff
@@ -655,14 +655,45 @@ async def dispatch(self, request: Request, call_next):
...
+                response.set_cookie(
+                    "jji_username",
+                    proxy_username,
+                    path="/",
+                    max_age=365 * 24 * 60 * 60,
+                    samesite="lax",
+                    secure=settings.secure_cookies,
+                )
+                return response
+            return await call_next(request)
```
[the remark this answers is not in the document]
Comment (myakove) [reply src/jenkins_job_insight/main.py:699]: Fixed as suggested.

**Comment by myakove** [comment]
@coderabbitai[bot]

The following review comments were reviewed and a decision was made:

### `src/jenkins_job_insight/main.py:638` (nitpick) — `638-647`: **Consider dropping `/register` from `_PUBLIC_PATHS`.
> Addressed: Fixed as suggested.
### `src/jenkins_job_insight/main.py:659` (nitpick) — `659-664`: **Parenthesize the mixed `and`/`or` for clarity (RUF021).
> Addressed: Fixed as suggested.

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai review

File: src/jenkins_job_insight/main.py [hunk 4]
```diff
@@ -655,14 +655,44 @@ async def dispatch(self, request: Request, call_next):
...
+                # SSO user hitting /register — redirect to dashboard
+                response = RedirectResponse(url="/", status_code=303)
+                response.set_cookie(
+                    "jji_username",
+                    proxy_username,
+                    path="/",
+                    max_age=365 * 24 * 60 * 60,
+                    samesite="lax",
+                    secure=settings.secure_cookies,
+                )
```
[the remark this answers is not in the document]
Comment (myakove) [reply src/jenkins_job_insight/main.py]: Fixed as suggested.

**Comment by myakove** [comment]
@coderabbitai review

**Comment by myakove** [comment]
@coderabbitai[bot]

The following review comments were reviewed and a decision was made:

### `src/jenkins_job_insight/main.py:686` (nitpick) — `686-693`: **Extract the duplicated `jji_username` set_cookie call.
> Addressed: Fixed as suggested.
### `src/jenkins_job_insight/main.py:673` (nitpick) — `673-695`: **Avoid the per-request session DB lookup on `/register` when no SSO header is present.
> Addressed: Fixed as suggested.

**Comment by myakove** [comment]
@coderabbitai review
